src/main/java/com/google/opensesame/user/UserData.java

```diff
@@ -0,0 +1,138 @@
+package com.google.opensesame.user;
+
+import com.google.opensesame.auth.AuthServlet;
+import com.google.opensesame.github.GitHubGetter;
+import java.io.IOException;
+import java.util.ArrayList;
+import org.kohsuke.github.GHUser;
+import org.kohsuke.github.GitHub;
+
+/**
+ * A fully fleshed out User of OpenSesame containing information sourced from GitHub. To conserve
+ * API calls, It would be best to only build UserData where neccesary. The main use of this class is
+ * to send data to the frontend as a JSON with Gson.
+ */
+public class UserData {
+  private Boolean isMentor;
+  private final ArrayList<String> interestTags;
+  private final ArrayList<String> menteeIDs;
+  private final ArrayList<String> projectIDs;
+  private final String bio;
+  private final String gitHubID;
+  private final String gitHubURL;
+  private final String image;
+  private final String location;
+  private final String name;
+  private final String userID;
+  private String email;
```

Comment:
Q: Is there a particular reason why these fields are private with getters instead of public?

Reply:
I wanted to avoid setting UserData, this might lead to issues where changes are made but never stored. This class is meant as a serialization too and nothing else.